packages/hooks/src/hooks/useTokenBalance.ts

```diff
@@ -33,25 +33,29 @@ export function useTokenBalance({ tokenAddress, accountAddress }: Props) {
   const [error, setError] = useState(false);
 
   const getBalance = async () => {
-    const contract = new ethers.Contract(tokenAddress, ERC20ABI, provider!);
-    const balance = await contract.balanceOf(accountAddress);
-    const decimals = await contract.decimals();
-    setBalance(balance);
-    setDecimals(decimals);
+    try {
+      const contract = new ethers.Contract(tokenAddress, ERC20ABI, provider!);
+      const balance = await contract.balanceOf(accountAddress);
+      const decimals = await contract.decimals();
+      setBalance(balance);
+      setDecimals(decimals);
+    } catch (err) {
+      setError(true);
+      setLoading(false);
+      console.error(error);
+    }
   };
 
   useEffect(() => {
-    if (tokenAddress && accountAddress && provider) {
-      setError(false);
-      setLoading(true);
-      try {
-        getBalance();
-      } catch (error) {
-        console.error(error);
-        setError(true);
-      }
+    if (!(tokenAddress && accountAddress && provider)) {
```

Comment:
@AlexNi245 can you describe why you made these changes? Once this is out of the way, we can merge.

Reply:
Yes, I made the changes to ensure that isLoading is set to false and isError is set to true if one of the 3 conditions in line 50 is not satisfied. In my opinion, it is a bit more readable to return if the conditions are not true rather than handle the error in the else block.